### PR TITLE
[TRAFODION-2598] Lob insert-select work from/to lob and external lob columns

### PR DESCRIPTION
**Pull request by sandhyasun**
Support for moving LOB data from source to target table containing LOB columns.

### CODE REVIEW

File: core/sql/generator/GenPreCode.cpp [hunk 1]
```diff
@@ -5570,18 +5570,36 @@ RelExpr * HbaseInsert::preCodeGen(Generator * generator,
...
+                                            << DgInt0((Int32)srcCol->lobStorageType())
+                                            << DgInt1((Int32)tgtCol->lobStorageType())
+                                            << DgString0(tgtCol->getColName());
                         GenExit();
                       }
-		  li->lobStorageLocation() = col->lobStorageLocation();
+                    }
+                    else
+                      if ((child1Expr->getOperatorType() == ITM_BASECOLUMN)||
+                          (child1Expr->getOperatorType() == ITM_INDEXCOLUMN))
```
Comment (DaveBirdsall) [inline core/sql/generator/GenPreCode.cpp]: Can a LOB column be an index column?

File: core/sql/exp/ExpLOBaccess.cpp [hunk 2]
```diff
@@ -2454,7 +2560,55 @@ Ex_Lob_Error ExLobsOper (
...
+    case Lob_InsSel:
+      {
+        
+        ExLob *srcLobPtr;
+        Int16 flags;
+        Lng32  lobNum;
+        Int32 lobType;
+        Int64 uid, inDescSyskey, descPartnKey;
+        short schNameLen;
+        char schName[1024];
```
Comment (DaveBirdsall) [inline core/sql/exp/ExpLOBaccess.cpp]: I see several fixed-length char arrays here. Do we have guarantees that these won't overrun?

File: core/sql/generator/GenPreCode.cpp (hunk 1 above)
Comment (sandhyasun) [inline core/sql/generator/GenPreCode.cpp]: This is restricted elsewhere in the binder.

**Comment by sandhyasun** [comment]
The rework includes replacing the fixed buffers with defines. Replaced several more occurences in various other places in the file.

**Pull request merged**